src/main/resources/assets/react/thesaurus/NewOrEditSynonym.tsx

```diff
@@ -168,186 +169,191 @@ export function NewOrEditSynonym(props) {
-													return <Table.Row key={`${i}.${j}`}>
-														<Table.Cell collapsing><EnonicCheckbox
-															checked={enabled}
-															path={`${pathASynonym}.enabled`}
-															type='radio'
-															toggle
-														/></Table.Cell>
-														<Table.Cell><EnonicInput
-															disabled={!enabled}
-															fluid
-															path={`${pathASynonym}.synonym`}
-															value={synonym}
-														/></Table.Cell>
-														<Table.Cell><EnonicInput
-															disabled={!enabled}
-															fluid
-															path={`${pathASynonym}.comment`}
-															value={comment}
-														/></Table.Cell>
-														<Table.Cell collapsing>
-															<Button.Group>
-																<InsertButton
-																	path={pathSynonyms}
-																	index={j+1}
-																	value={{
-																		comment: '',
-																		enabled: true,
-																		synonym: ''
-																	}}
-																/>
-																<MoveDownButton
-																	disabled={j + 1 >= synonyms.length}
-																	path={pathSynonyms}
-																	index={j}
-																/>
-																<MoveUpButton
-																	path={pathSynonyms}
-																	index={j}
-																/><DeleteItemButton
-																	disabled={synonyms.length < 2}
-																	path={pathSynonyms}
-																	index={j}
-																/>
-															</Button.Group>
-														</Table.Cell>
-													</Table.Row>;
-												});
-											})()}
-										</Table.Body>
-										{/*<Table.Footer>
-											<Table.HeaderCell colspan={4}>
-												<Button.Group>
-													<InsertButton
-														path={pathSynonyms}
-														index={synonyms.length}
-														value={{
-															comment: '',
-															enabled: true,
-															synonym: ''
-														}}
-													/>
-												</Button.Group>
-											</Table.HeaderCell>
-										</Table.Footer>*/}
-									</Table>
-									{/*<SortButton path={pathSynonyms}/>*/}
-									<Button.Group floated='right'>
-										<MoveDownButton
-											disabled={i + 1 >= languages.length}
-											path={PATH_LANGUAGES}
-											index={i}
-										/>
-										<MoveUpButton
-											path={PATH_LANGUAGES}
-											index={i}
-										/>
-										<DeleteItemButton
-											disabled={languages.length < 2}
-											icon={true}
-											path={PATH_LANGUAGES}
-											index={i}
-										><Icon color='red' name='alternate outline trash'/> Delete language {language}</DeleteItemButton>
-									</Button.Group>
-									<div style={{clear:'both'}}></div>
-								</Panel>
-							</Segment>;
-						})}
-						<Segment>
-							<InsertButton
-								icon={false}
-								index={languages.length}
-								path={PATH_LANGUAGES}
-								value={{
-									language: '',
+				{ENABLE_EXPERIMENTAL_GUI
```

Comment:
No, this was a whole GUI I tried out, to see how it could be done, I never got time to convince Thomas, but this should in some variant be used in some future feature release of Explorer.

Reply:
I guess it would look nicer, if I simply moved that code out, to it's own component.